eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -236,8 +236,49 @@ public function getCriterion(APILimitationValue $value, APIUserReference $curren
             return new Criterion\ObjectStateId($value->limitationValues[0]);
         }
 
-        // several limitation values: IN operation
-        return new Criterion\ObjectStateId($value->limitationValues);
+        $groupedLimitationValues = $this->groupLimitationValues($value->limitationValues);
+
+        if (count($groupedLimitationValues) === 1) {
+            // several limitation values: IN operation
+            return new Criterion\ObjectStateId($value->limitationValues);
```

Comment:
@natanael89 so are you saying this check could have been done before call to `groupLimitationValues ` then?

as a:
```php
    if (count($value->limitationValues) === 1) {
       // several limitation values: IN operation
       return new Criterion\ObjectStateId($value->limitationValues);
    }

```